src/test/scala/latis/input/HapiBinaryAdapterSpec.scala

```diff
@@ -19,17 +20,18 @@ class HapiBinaryAdapterSpec extends AnyFlatSpec {
   private lazy val uri: URI = FileUtils.resolvePath("data/hapi_binary_data").get.toUri
 
   private lazy val reader = new AdaptedDatasetReader {
+
     def model: DataType = Function.from(
-      Scalar.fromMetadata(Metadata("id" -> "Time", "type" -> "string", "length" -> "24")).value,
+      Time.fromMetadata(Metadata("id" -> "Time", "type" -> "string", "size" -> "24", "units"->"yyyy-MM-dd")).value,
```

Comment:
Should these units reflect the 24-character format of the time values? I wonder if we need to beef up `Time.fromMetadata` so it won't allow a size that is incompatible with the units.

Reply:
I would note that [here](https://github.com/lasp/hapi-server/commit/f20a211ab936ff191f953fcebf2fed1ced83f3ad) we enforced that the HAPI server always use a size of 24 for time, and other unit-types should get converted/expanded to "yyyy-MM-ddThh:mm:ss.sssZ" which will always take 24. I figured it would be safe to assume that the data we read from HAPI would always be in the 24 sized format, and that the Time.fromMetadata with "yyyy-MM-dd" would correctly match to it. I can change that to explicitly be the longer unit description, if that makes sense.

Reply:
The binary data file contains times like `2019-01-01T00:00:00.000Z` so the `units` should match that. Otherwise I would expect some spooky results somewhere.